flight/UAVObjects/uavobjectmanager.c

```diff
@@ -1658,24 +1681,58 @@ void UAVObjIterate(void (*iterator) (UAVObjHandle obj))
 	PIOS_Recursive_Mutex_Unlock(mutex);
 }
 
-/**
- * Send a triggered event to all event queues registered on the object.
- */
-static int32_t sendEvent(struct UAVOBase * obj, uint16_t instId,
-			UAVObjEventType triggered_event)
-{
-	/* Set up the message that will be sent to all registered listeners */
-	UAVObjEvent msg = {
-		.obj    = (UAVObjHandle) obj,
-		.event  = triggered_event,
-		.instId = instId,
-	};
+/* type signature must match invokeCallback below, with 4 or fewer args */
+static void __attribute__((used)) realInvokeCallback(struct ObjectEventEntry *event,
+		UAVObjEvent *msg, void *obj_data, int len);
 
+static void realInvokeCallback(struct ObjectEventEntry *event,
+		UAVObjEvent *msg, void *obj_data, int len) {
+	event->cb(msg, event->cbInfo.cbCtx, obj_data, len);
+}
+
+#if (!defined(SIM_POSIX)) && defined(__arm__)
+static void invokeCallback(struct ObjectEventEntry *event, UAVObjEvent *msg,
```

Comment:
I'll try -- I've tried to put in everything relevant that I could think of, but it's worth another pass to see if anything more can be noted.

Also if there's anything particularly crazy looking let me know and I can explain / comment that in particular.